src/main/java/com.gigssandbox/UserInteraction.java

```diff
@@ -7,89 +7,85 @@
 
 import java.util.Collection;
 
-public class UserInteraction {
+class UserInteraction {
 
-    private GigsRepository repository;
-    private ConsoleHelper helper;
-    private DataProcessor processor;
+    private GigsRepository gigsRepository;
+    private UserMessenger userMessenger;
+    private DataProcessor dataProcessor;
 
-    public UserInteraction() {
-        repository = new GigsRepository();
-        helper = new ConsoleHelper();
-        processor = new DataProcessor();
+    UserInteraction() {
+        this.gigsRepository = new GigsRepository();
+        this.userMessenger = new UserMessenger();
+        this.dataProcessor = new DataProcessor();
     }
 
-    void interact() {
-        int actionCode;
+    void startInteraction() {
+        userMessenger.sendMessage("hello");
 
-        helper.writeStringFromPropertiesToConsole("hello");
-        helper.writeStringFromPropertiesToConsole("choose_action");
-
-        while ((actionCode = helper.readInt()) > 0) {
-            switch (actionCode) {
-                case 1:
+        Command currentCommand = userMessenger.nextCommand();
+        while (!currentCommand.equals(Command.EXIT)) {
+            switch (currentCommand) {
+                case PRINT_GIGS:
                     printGigsCollection(1);
                     break;
-                case 2:
+                case PRINT_GIGS_BY_BAND:
                     printGigsCollection(2);
                     break;
-                case 3:
+                case ADD_GIG:
                     insertGig();
                     break;
-                case 4:
+                case ADD_BAND:
                     insertBand();
                     break;
-                case 5:
-                    helper.writeStringFromPropertiesToConsole("say_goodbye");
-                    System.exit(0);
-                default:
-                    helper.writeStringFromPropertiesToConsole("unsupported");
-                    break;
+                case UNSUPPORTED:
+                    userMessenger.sendMessage("unsuported");
             }
-            helper.writeStringFromPropertiesToConsole("choose_action");
+            userMessenger.sendMessage("choose_action");
+            currentCommand = userMessenger.nextCommand();
```

Comment:
I think we can inline getting next command: "while((Command command = userMessenger.nextCommand()) != ...)"

Reply:
![2](https://user-images.githubusercontent.com/12383964/50539167-c205d580-0b8c-11e9-871a-f84a497caf53.jpg)
it doesn't work in such way

Reply:
![while_command](https://user-images.githubusercontent.com/45514597/50539231-ddbdab80-0b8d-11e9-90cc-33a06805b5a3.jpg)

lets try like this

Reply:
Enums can be compared on equality by either "==" or "equals", doesnt matter. I think "==" is faster to read.